### PR TITLE
Update SCE file contents to include processed SCE objects

### LINKED ISSUE #93: Document contents for filtered and normalized SCE objects
The `sce_file_contents.md` docs should be updated with any additional information that is included in the `_processed.rds` object that will now be included for download on the portal. In particular, the following were added to the metadata:
`ccdl_filter_method`
`prob_compromised_cutoff`
`min_gene_cutoff`
`normalization`
`highly_variable_genes`

There is also the addition of the `ccdl_filter` column in the `colData`, and we should note that `rowData` statistics have been recalculated after filtering.

### PR DESCRIPTION
**Pull request by allyhawkins**
Closes #93. 

This PR updates the `sce_file_contents.md` section to include mention of the processed SCE object and any items that can be specifically found in the processed SCE object. 

I made a few updates throughout here: 

- I added a section below the `colData` tables discussing the presence of the `ccdl_filter` column that will be found in the processed objects. I mentioned briefly here that we do filtering, but not how the criteria are chosen. Instead I direct them to the processing information. The point I wanted to get across here is that there will be an additional column in the `colData` but they should all be labeled as `Keep` if they are in the processed object. 
- I added any of the new changes that have been made to the metadata list for both the filtered objects and then the processed objects. I just added to the list and made a note if they were only found in the processed object. 
- I added a new section for the dimensionality results. I kept it fairly short, but I didn't think it fit in anywhere else so included a brief discussion of what was calculated and how they can be accessed.

### CODE REVIEW

File: docs/sce_file_contents.md [hunk 1]
```diff
@@ -53,6 +53,12 @@ These metrics were calculated by using [`miQC`](https://bioconductor.org/package
 | `prob_compromised`      | Probability that a cell is compromised (i.e., dead or damaged), as calculated by `miQC`                                                                                                       |
 | `miQC_pass`             | Indicates whether the cell passed the default miQC filtering. `TRUE` is assigned to cells with a low probability of being compromised (`prob_compromised` < 0.75) or [sufficiently low mitochondrial content](https://bioconductor.org/packages/release/bioc/vignettes/miQC/inst/doc/miQC.html#preventing-exclusion-of-low-mito-cells).  |
 
+For the `_processed.rds` file only, an additional column named, `ccdl_filter` will be included in the `colData`.
```
Comment (jashapiro) [inline docs/sce_file_contents.md:56]: Did we decide not to add this back to the filtered data? I thought that we had, according to: 
https://github.com/AlexsLemonade/scpca-nf/blob/f797157dff44804500774f164cc518017c0871e8/bin/post_process_sce.R#L147-L148
and 
https://github.com/AlexsLemonade/scpca-nf/blob/f797157dff44804500774f164cc518017c0871e8/modules/sce-processing.nf#L126-L134

File: docs/sce_file_contents.md [hunk 2]
```diff
@@ -94,7 +100,31 @@ expt_metadata <- metadata(sce)
...
+| `umi_cutoff`        | The minimum UMI count per cell used as a threshold for removing empty droplets. Only present for `filtered` objects where the `filtering_method` is `UMI cutoff` |
+| `prob_compromised_cutoff`        | The minimum cutoff for the probability of a cell being compromised, as calculated by `miQC`. Only present for `filtered` objects |
+| `ccdl_filter_method`        | Method used by the Data Lab to filter low quality cells prior to normalization. Either `miQC` or `Minimum_gene_cutoff`. Only present for `processed` objects |
+| `min_gene_cutoff`        | The minimum cutoff for the number of unique genes detected per cell. Only present for `processed` objects |
+| `normalization`        | The method used for normalization of raw counts. Either `deconvolution`, described in [Lun, Bach, and Marioni (2016)](https://doi.org/10.1186/s13059-016-0947-7), or  `log-normalization`. Only present for `processed` objects |
+| `highly_variable_genes`        | A list of highly variable genes detected in the library, determined using `scran::modelGeneVar` and `scran::getTopHVGs`. Only present for `processed` objects |
```
Comment (jashapiro) [inline docs/sce_file_contents.md:108]: I don't think we want to say that we "determined" the HVGs, more how they were used. They are not a definitive list, to be sure
```suggestion
| `highly_variable_genes`        | A list of highly variable genes used for dimensionality reduction, determined using `scran::modelGeneVar` and `scran::getTopHVGs`. Only present for `processed` objects |
```

File: docs/sce_file_contents.md [hunk 3]
```diff
@@ -94,7 +100,31 @@ expt_metadata <- metadata(sce)
...
 | `miQC_model`        | The model object that `miQC` fit to the data and was used to calculate `prob_compromised`. Only present for `filtered` objects |
 | `filtering_method`  | The method used for cell filtering. One of `emptyDrops`, `emptyDropsCellRanger`, or `UMI cutoff`. Only present for `filtered` objects |
-| `umi_cutoff`        | The minimum UMI count per cell used as a threshold for filtering. Only present for `filtered` objects where the `filtering_method` is `UMI cutoff` |
+| `umi_cutoff`        | The minimum UMI count per cell used as a threshold for removing empty droplets. Only present for `filtered` objects where the `filtering_method` is `UMI cutoff` |
+| `prob_compromised_cutoff`        | The minimum cutoff for the probability of a cell being compromised, as calculated by `miQC`. Only present for `filtered` objects |
+| `ccdl_filter_method`        | Method used by the Data Lab to filter low quality cells prior to normalization. Either `miQC` or `Minimum_gene_cutoff`. Only present for `processed` objects |
+| `min_gene_cutoff`        | The minimum cutoff for the number of unique genes detected per cell. Only present for `processed` objects |
```
Comment (jashapiro) [inline docs/sce_file_contents.md:106]: I believe these are both in the filtered object as well.
https://github.com/AlexsLemonade/scpca-nf/blob/f797157dff44804500774f164cc518017c0871e8/bin/post_process_sce.R#L68-L85

File: docs/sce_file_contents.md [hunk 4]
```diff
@@ -94,7 +100,31 @@ expt_metadata <- metadata(sce)
...
+| `normalization`        | The method used for normalization of raw counts. Either `deconvolution`, described in [Lun, Bach, and Marioni (2016)](https://doi.org/10.1186/s13059-016-0947-7), or  `log-normalization`. Only present for `processed` objects |
+| `highly_variable_genes`        | A list of highly variable genes detected in the library, determined using `scran::modelGeneVar` and `scran::getTopHVGs`. Only present for `processed` objects |
+
+### Dimensionality reduction results
+
+In the RDS file containing the processed `SingleCellExperiment` object only (`_processed.rds`), the `reducedDim` slot of the object will be occupied with both principal component analysis (PCA) and UMAP results.
```
Comment (jashapiro) [inline docs/sce_file_contents.md:112]: I added quotes here as these are the actual names in the slots. 
```suggestion
In the RDS file containing the processed `SingleCellExperiment` object only (`_processed.rds`), the `reducedDim` slot of the object will be occupied with both principal component analysis (`PCA`) and `UMAP` results.
```

File: docs/sce_file_contents.md [hunk 5]
```diff
@@ -94,7 +100,31 @@ expt_metadata <- metadata(sce)
...
+For all other files, the `reducedDim` slot will be empty as no dimensionality reduction was performed.
+
+PCA results were calculated using `scater::runPCA()`, specifying to only use the highly variable genes stored in the object in `metadata(sce)$highly_variable_genes`.
+The below command can be used to access the PCA results:
```
Comment (jashapiro) [inline docs/sce_file_contents.md:116]:
```suggestion
The following command can be used to access the PCA results:
```

File: docs/sce_file_contents.md [hunk 6]
```diff
@@ -94,7 +100,31 @@ expt_metadata <- metadata(sce)
...
+```r
+reducedDim(sce, "PCA")
+```
+
+UMAP results were calculated using `scater::runUMAP()`, with the PCA results as input rather than the full gene expression matrix.
+The below command can be used to access the UMAP results:
+
+```r
+reducedDim(sce,"UMAP")
+```
```
Comment (jashapiro) [inline docs/sce_file_contents.md:127]:
```suggestion
The following command can be used to access the UMAP results:

```r
reducedDim(sce, "UMAP")
```
```

File: docs/sce_file_contents.md [hunk 7]
```diff
@@ -94,7 +100,31 @@ expt_metadata <- metadata(sce)
...
+| `min_gene_cutoff`        | The minimum cutoff for the number of unique genes detected per cell. Only present for `processed` objects |
+| `normalization`        | The method used for normalization of raw counts. Either `deconvolution`, described in [Lun, Bach, and Marioni (2016)](https://doi.org/10.1186/s13059-016-0947-7), or  `log-normalization`. Only present for `processed` objects |
+| `highly_variable_genes`        | A list of highly variable genes detected in the library, determined using `scran::modelGeneVar` and `scran::getTopHVGs`. Only present for `processed` objects |
+
+### Dimensionality reduction results
+
+In the RDS file containing the processed `SingleCellExperiment` object only (`_processed.rds`), the `reducedDim` slot of the object will be occupied with both principal component analysis (PCA) and UMAP results.
+For all other files, the `reducedDim` slot will be empty as no dimensionality reduction was performed.
+
+PCA results were calculated using `scater::runPCA()`, specifying to only use the highly variable genes stored in the object in `metadata(sce)$highly_variable_genes`.
```
Comment (jashapiro) [inline docs/sce_file_contents.md:115]: Do we want to specify the number of PCs (I don't think we need to, but I was thinking about it)
```suggestion
PCA results were calculated using `scater::runPCA()`, using only highly variable genes.
The list of highly variable genes used was selected using `scran::modelGeneVar` and `scran::getTopHVGs`, and is stored in the SCE object in `metadata(sce)$highly_variable_genes`.
```

**Review by jashapiro — COMMENTED** [review]
This looks good; I had some questions about which files have which slots: I think that some should end up back in the filtered data, but I want to check on that!

File: docs/sce_file_contents.md (hunk 1 above)
Comment (allyhawkins) [reply docs/sce_file_contents.md:56]: Yes you are right! I'll update this.

**Comment by allyhawkins** [comment]
I went ahead and updated this to include the correct contents for the `filtered` objects. I had forgotten that we re-write out the filtered object with the additional column and metadata. This should be ready for another look.

**Review by jashapiro — COMMENTED** [review]
Looks good; thanks for the updates. I think we need to also update lines 48-49 to mention that these columns are present in `processed` objects as well, and maybe reword to include something about our filtering  as well?

**Comment by allyhawkins** [comment]
I added another line that includes the filtering cutoffs and noted that the columns should be found in both the `filtered` and `processed` objects. Let me know if you had something else in mind.

File: docs/sce_file_contents.md [hunk 8]
```diff
@@ -45,15 +45,16 @@ The following per-cell data columns are included for each cell, calculated using
 | `subsets_mito_percent`  | Percent of all UMI counts assigned to mitochondrial genes                                                                                                                                     |
 | `total`                 | Total UMI count for RNA-seq data and any alternative experiments (i.e., CITE-seq)                                                                                                             |
 
-The following are additional per-cell data columns included only in `filtered` objects.
+The following are additional per-cell data columns included in both the `filtered` and `processed` objects.
 These metrics were calculated by using [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html), a package that jointly models proportion of reads belonging to mitochondrial genes and number of unique genes detected to predict low-quality cells.
+Cells with a high likelihood of being compromised as calculated by `miQC` (greater than 0.75) and cells that do not pass a minimum number of unique genes detected threshold of 200 are removed from the counts matrix found in the `processed` objects.
```
Comment (jashapiro) [inline docs/sce_file_contents.md:50]: I reworded somewhat. I'm not sure we need the detail about the filtering here: I moved the link to the processed gene expression data section up here for brevity. (So line 59 should be deleted)

Secondary thought: we should rename `ccdl_filter` to `scpca_filter`.

```suggestion
These include metrics were calculated by [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html), a package that jointly models proportion of reads belonging to mitochondrial genes and number of unique genes detected to predict low-quality cells.
We also include the filtering results used for the creation of the `processed` objects. 
See the description of the {ref}`processed gene expression data <processing_information:Processed gene expression data>` for more information on filtering performed to obtain the `_processed.rds` file.
```

Comment (jashapiro) [reply docs/sce_file_contents.md]: Filed https://github.com/AlexsLemonade/scpca-nf/pull/239 for the secondary thought... We should update all docs here if that is approved.

File: docs/sce_file_contents.md [hunk 9]
```diff
@@ -45,13 +45,17 @@ The following per-cell data columns are included for each cell, calculated using
 | `subsets_mito_percent`  | Percent of all UMI counts assigned to mitochondrial genes                                                                                                                                     |
 | `total`                 | Total UMI count for RNA-seq data and any alternative experiments (i.e., CITE-seq)                                                                                                             |
 
-The following are additional per-cell data columns included only in `filtered` objects.
-These metrics were calculated by using [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html), a package that jointly models proportion of reads belonging to mitochondrial genes and number of unique genes detected to predict low-quality cells.
+The following are additional per-cell data columns included in both the `filtered` and `processed` objects.
+The included metrics were calculated by [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html), a package that jointly models proportion of reads belonging to mitochondrial genes and number of unique genes detected to predict low-quality cells.
```
Comment (jashapiro) [inline docs/sce_file_contents.md:49]: My initial edit here was intended to make the statement about metrics less definitive about the source of the data, but I left in the "were"... But then I realize that even with my intended change, that left an unqualified "these", so here I am attempting once again to say what I mean.

```suggestion
The following additional per-cell data columns are included in both the `filtered` and `processed` objects.
These columns include metrics calculated by [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html), a package that jointly models proportion of reads belonging to mitochondrial genes and number of unique genes detected to predict low-quality cells.
```

**Review by jashapiro — APPROVED** [review]
This looks good. I had one place where my previous edit wasn't quite complete, so I am trying  again... It is minor though, so approving either way.
